components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java

```diff
@@ -762,13 +786,19 @@ public AccessTokenDO retrieveAccessToken(String accessTokenIdentifier, boolean i
                 OAuth2Util.checkUserNameAssertionEnabled()) {
                 userStoreDomain = OAuth2Util.getUserStoreDomainFromAccessToken(accessTokenIdentifier);
             }
-            String sql = SQLQueries.RETRIEVE_ACTIVE_ACCESS_TOKEN;
+
+            String sql;
+
             if (includeExpired) {
                 sql = SQLQueries.RETRIEVE_ACTIVE_EXPIRED_ACCESS_TOKEN;
+            } else {
+                sql = SQLQueries.RETRIEVE_ACTIVE_ACCESS_TOKEN;
             }
```

Comment:
This is due to the strange behavior of the string replacement. I don't know whether it is a isolated incident in my PC or a JVM issue.